ts/src/delta.ts

```diff
@@ -229,6 +230,108 @@ export default class delta extends Exchange {
+        // support expired option contracts
+        const quote = 'USDT';
+        const optionParts = symbol.split ('-');
+        const symbolBase = symbol.split ('/');
+        let base = undefined;
+        let expiry = undefined;
+        let optionType = undefined;
+        if (symbol.indexOf ('/') > -1) {
+            base = this.safeString (symbolBase, 0);
+            expiry = this.safeString (optionParts, 1);
+            optionType = this.safeString (optionParts, 3);
+        } else {
+            base = this.safeString (optionParts, 1);
+            expiry = this.safeString (optionParts, 3);
+            optionType = this.safeString (optionParts, 0);
+        }
+        const settle = quote;
+        const strike = this.safeString (optionParts, 2);
+        const datetime = this.convertExpireDate (expiry);
+        const timestamp = this.parse8601 (datetime);
+        return {
+            'id': optionType + '-' + base + '-' + strike + '-' + expiry,
+            'symbol': base + '/' + quote + ':' + settle + '-' + expiry + '-' + strike + '-' + optionType,
+            'base': base,
+            'quote': quote,
+            'settle': settle,
+            'baseId': base,
+            'quoteId': quote,
+            'settleId': settle,
+            'active': false,
+            'type': 'option',
+            'linear': undefined,
+            'inverse': undefined,
+            'spot': false,
+            'swap': false,
+            'future': false,
+            'option': true,
+            'margin': false,
+            'contract': true,
+            'contractSize': this.parseNumber ('1'),
+            'expiry': timestamp,
+            'expiryDatetime': datetime,
+            'optionType': (optionType === 'C') ? 'call' : 'put',
+            'strike': this.parseNumber (strike),
+            'precision': {
+                'amount': undefined,
+                'price': undefined,
+            },
+            'limits': {
+                'amount': {
+                    'min': undefined,
+                    'max': undefined,
+                },
+                'price': {
+                    'min': undefined,
+                    'max': undefined,
+                },
+                'cost': {
+                    'min': undefined,
+                    'max': undefined,
+                },
+            },
+            'info': undefined,
+        };
+    }
+
+    market (symbol) {
+        if (this.markets === undefined) {
+            throw new ExchangeError (this.id + ' markets not loaded');
+        }
+        if (typeof symbol === 'string') {
+            if (symbol in this.markets) {
+                return this.markets[symbol];
+            } else if (symbol in this.markets_by_id) {
+                const markets = this.markets_by_id[symbol];
+                return markets[0];
+            } else if ((symbol.indexOf ('-C') > -1) || (symbol.indexOf ('-P') > -1) || (symbol.indexOf ('C')) || (symbol.indexOf ('P'))) {
```

Comment:
@Dan-krm This logic isn't good for a few reasons.

- first, symbol.indexOf ('C') evaluates to `-1` so the boolean version of this is true, so this evaluates to true whenever the letter `C` or `P` is not present, but it also evaluates to true whenever C/P is at any index other than 0, the only time `symbol.indexOf ('C')` evaluates to false is when C is the first letter of symbol, but then because we also have `symbol.indexOf ('P')` right after, this line will always evaluate to true, because the first letter of `symbol` can't be both `C` and `P`.

you can test this in node

```
const symbol = "market";
undefined
> Boolean(symbol.indexOf ('P'))
true
```

- second, you can't just test for the presence of `C` or `P` to see if the symbol is an option symbol, by this logic, this symbol would be an option symbol `ETH/BTC`

--------------

can you update this line to use some logic that is correct? Thanks

Reply:
@samgermain Yeah definitely, I'll adjust this

Reply:
#20634